src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -428,6 +431,19 @@ def module_is_offloaded(module):
                 f"It seems like you have activated model offloading by calling `enable_model_cpu_offload`, but are now manually moving the pipeline to GPU. It is strongly recommended against doing so as memory gains from offloading are likely to be lost. Offloading automatically takes care of moving the individual components {', '.join(self.components.keys())} to GPU when needed. To make sure offloading works as expected, you should consider moving the pipeline back to CPU: `pipeline.to('cpu')` or removing the move altogether if you use offloading."
             )
 
+        pipeline_has_bnb = any(any((_check_bnb_status(module))) for _, module in self.components.items())
+        # PR: https://github.com/huggingface/accelerate/pull/3223/
+        if (
+            not pipeline_is_offloaded
+            and not pipeline_is_sequentially_offloaded
+            and pipeline_has_bnb
+            and torch.device(device).type == "cuda"
+            and is_accelerate_version("<", "1.1.0.dev0")
+        ):
+            raise ValueError(
```

Comment:
the error message you want to throw against this scenario, no?
1. accelerator < 1.1.0.dev0
2. `you call pipeline.to("cuda")` on a pipeline that has bnb 

but if these 2 condition are met (older accelerator version + bnb):
1. not pipeline_is_sequentially_offloaded` will be `False` here and you will not reach the value error 
2. you will reach this check first and get an error message -this is the wrong error message I was talking about https://github.com/huggingface/diffusers/blob/8421c1461bf4ab7801070d04d6ec1e6b28ee5b59/src/diffusers/pipelines/pipeline_utils.py#L413

```python
 if (
            not pipeline_is_offloaded
            and not pipeline_is_sequentially_offloaded
            and pipeline_has_bnb
            and torch.device(device).type == "cuda"
            and is_accelerate_version("<", "1.1.0.dev0")
        ):
```